File: executable/python/axes-range-mode.py
import plotly.plotly as py
from plotly.graph_objs import *

py.sign_in('test-runner', '9h29fe3l0x')

data = Data([
    Scatter(
        x=[2, 4, 6],
        y=[-3, 0, 3]
    )
])
layout = Layout(
    xaxis=XAxis(
        autorange=True,
        rangemode='tozero'
    ),
    yaxis=YAxis(
        autorange=True,
        rangemode='nonnegative'
    ),
    showlegend=False
)
fig = Figure(data=data, layout=layout)

plot_url = py.plot(fig, filename='axes-range-mode')

File: executable/r/axes-range-mode.r
library(plotly)

p <- plotly(username='test-runner', key='9h29fe3l0x')

trace1 <- list(
  x = c(2, 4, 6), 
  y = c(-3, 0, 3), 
  type = "scatter"
)

layout <- list(
  xaxis = list(
    autorange = TRUE, 
    rangemode = "tozero"
  ), 
  yaxis = list(
    autorange = TRUE, 
    rangemode = "nonnegative"
  ), 
  showlegend = FALSE
)



response <- p$plotly(trace0, kwargs=list(layout=layout, filename="axes-range-mode", fileopt="overwrite"))
url <- response$url
filename <- response$filename